### PR TITLE
test: Modify common to support Firecracker

### LINKED ISSUE #1070: Modify common to support Firecracker
We need to modify the common.bash in order to check for the Firecracker
path as well to check the processes that run with Firecracker and Kata.

### PR DESCRIPTION
**Pull request by GabyCT**
We need to modify the common.bash in order to check for the Firecracker
path as well to check the processes that run with Firecracker and Kata.

Fixes #1070

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: lib/common.bash [hunk 1]
```diff
@@ -73,7 +79,11 @@ extract_kata_env(){
 # Checks that processes are not running
 check_processes() {
 	extract_kata_env
-	general_processes=( ${PROXY_PATH} ${HYPERVISOR_PATH} ${SHIM_PATH} )
+	if [ "$KATA_HYPERVISOR" == firecracker ]; then
```
Comment (chavafg) [inline lib/common.bash]: I'll prefer to make this more generic as this is the case when `vsock` is enabled.
What about check if `vsock` is enabled and then do not check for the proxy process. 
You can use this: https://github.com/kata-containers/tests/blob/master/integration/stability/soak_parallel_rm.sh#L64-L72

**Comment by GabyCT** [comment]
@chavafg changes applied

File: lib/common.bash [hunk 2]
```diff
@@ -73,7 +79,12 @@ extract_kata_env(){
 # Checks that processes are not running
 check_processes() {
 	extract_kata_env
-	general_processes=( ${PROXY_PATH} ${HYPERVISOR_PATH} ${SHIM_PATH} )
+	vsock_configured=$($RUNTIME_PATH kata-env | awk '/UseVSock/ {print $3}')
```
Comment (chavafg) [inline lib/common.bash]: you should also check if `vsock` is supported on the environment as sometimes the setting is enabled in the config file,  but the `vsock` module is not loaded. 
`vsock_supported=$($RUNTIME_PATH kata-env | awk '/SupportVSock/ {print $3}')`

**Comment by GabyCT** [comment]
@chavafg changes applied

**Comment by grahamwhaley** [comment]
We got one test fail on the vsocks - I expect unrelated, but I'll nudge the job to retry:

```
03:22:59 ------------------------------
03:22:59 package manager update test check apt-get update 
03:22:59   should not fail
03:22:59   /tmp/jenkins/workspace/kata-containers-tests-fedora-vsocks-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:38
03:22:59 Running command '/usr/bin/docker [docker run --cidfile /tmp/cid136896127/TGY8Vw26AlUet7qThSfl6OSr4qNypu --runtime kata-runtime --rm --name TGY8Vw26AlUet7qThSfl6OSr4qNypu debian apt-get -y update]'
03:23:59 Killing process timeout reached '60' seconds
03:23:59 Running command '/usr/bin/docker [docker ps -a -f name=TGY8Vw26AlUet7qThSfl6OSr4qNypu --format {{.Status}}]'
03:23:59 [docker ps -a -f name=TGY8Vw26AlUet7qThSfl6OSr4qNypu --format {{.Status}}]
03:23:59 Timeout: 60 seconds
03:23:59 Exit Code: 0
03:23:59 Stdout: Up 56 seconds
03:23:59 
03:23:59 Stderr: 
03:23:59 
03:23:59 • Failure [60.077 seconds]
03:23:59 package manager update test
03:23:59 /tmp/jenkins/workspace/kata-containers-tests-fedora-vsocks-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:15
03:23:59   check apt-get update
03:23:59   /tmp/jenkins/workspace/kata-containers-tests-fedora-vsocks-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:37
03:23:59     should not fail [It]
03:23:59     /tmp/jenkins/workspace/kata-containers-tests-fedora-vsocks-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:38
03:23:59 
03:23:59     Expected
03:23:59         <int>: -1
03:23:59     to be zero-valued
03:23:59 
03:23:59     /tmp/jenkins/workspace/kata-containers-tests-fedora-vsocks-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:41
```

**Pull request merged**